manipulator:
  joint_names: ['shoulder_rotation', 'forearm_rotation', 'arm_rotation', 'wrist_rotation', 'wrist_pitch', 'gripper_rotation', ]

# Joint states controller settings
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50


# Joint position controllers settings
  shoulder_position_controller:
    type: velocity_controllers/JointPositionController
    joint: shoulder_rotation
    pid: {p: 2.0, i: 1.0, d: 0.0}

  forearm_position_controller:
    type: velocity_controllers/JointPositionController
    joint: forearm_rotation
    pid: {p: 200.0, i: 40.0, d: 0.0}

  arm_position_controller:
    type: velocity_controllers/JointPositionController
    joint: arm_rotation
    pid: {p: 50.0, i: 10.0, d: 0.0}

  wrist_rot_position_controller:
    type: velocity_controllers/JointPositionController
    joint: wrist_rotation
    pid: {p: 1.5, i: 0.5, d: 0.0}

  wrist_pitch_position_controller:
    type: velocity_controllers/JointPositionController
    joint: wrist_pitch
    pid: {p: 500.0, i: 5.0, d: 0.01}

  gripper_position_controller:
    type: velocity_controllers/JointPositionController
    joint: gripper_rotation
    pid: {p: 800.0, i: 50.0, d: 0.01}


  gazebo_ros_control:
    pid_gains:
      shoulder_rotation:
        p: 2.0
        i: 1.0
        d: 0.0
      forearm_rotation:
        p: 200.0
        i: 40.0
        d: 0
      arm_rotation:
        p: 50.0
        i: 10.0
        d: 0.0
      wrist_rotation:
        p: 1.5
        i: 0.5
        d: 0.0
      wrist_pitch:
        p: 20.0
        i: 5.0
        d: 0.0
      gripper_rotation:
        p: 0.5
        i: 1.0
        d: 0.0